### PR TITLE
Add basic application running

### PR DESCRIPTION
**Pull request by phanphare**
Use QProcess to run applications. At the moment, it only allows for one application at a time to be running, and the running application runs in its own thread but can be asynchronously accessed by the launcher. On OS X, it searchs the .app/Contents/Mac OS/ folder for the binary, and on other platforms it simply runs the executable the user has selected.

### CODE REVIEW

File: Source/Library.cpp [hunk 1]
```diff
@@ -23,18 +29,32 @@ void Library::on_testLaunch_clicked()
...
     // TODO: Implement some threading
-    QProcess process(this);
-    process.start("\"" + location + "\"");
-    process.waitForFinished(-1);
+    if (!processRunning) {
+        runningProcess->start(location);
+        runningProcess->waitForStarted();
+        processRunning = true;
+    } else {
+        QMessageBox messageBox(this);
```
Comment (lolzballs) [inline Source/Library.cpp]: Putting the library class as the parent will cause the QPushButtons on the QMessageBox to be styled, which is not what is desired.

Comment (lolzballs) [inline Source/Library.cpp]: Also, you probably want to display the message box before the QFileDialog pops up to select.

**Comment by phanphare** [comment]
I've fixed the issues lolzballs pointed out; thanks for the feedback! The syntax error is also now fixed.

File: Source/Library.cpp [hunk 2]
```diff
@@ -10,31 +11,53 @@ Library::Library(QWidget *parent) :
...
-    dialog.setFileMode(QFileDialog::ExistingFile);
-
-
-    if (dialog.exec()) {
-        QStringList files = dialog.selectedFiles();
-        QString file = files.at(0);
 
-        runProcess(file);
+    if (!processRunning) {
+        QFileDialog dialog();
```
Comment (Hal9007) [inline Source/Library.cpp]: I'm getting build errors on both the initialization of the QFileDialog and the QMessageBox below, both in which the compiler thinks you're declaring a function "dialog()" with return type QFileDialog. You should specify the constructor by adding pointers to the parent widget in the initialization of each child.

Comment (phanphare) [inline Source/Library.cpp]: Okay, thanks. I'm not sure how I ended up pushing this commit without compiling first...

**Comment by Hal9007** [comment]
Thanks for fixing that up, phanphare. Build looks good on Windows and Linux.

**Pull request merged**